miden-crypto/src/dsa/rpo_falcon512/keys/public_key.rs

```diff
@@ -23,63 +15,38 @@ use crate::{Word, dsa::rpo_falcon512::FALCON_ENCODING_BITS};
 // PUBLIC KEY
 // ================================================================================================
 
-/// A public key for verifying signatures.
-///
-/// The public key is a [Word] (i.e., 4 field elements) that is the hash of the coefficients of
-/// the polynomial representing the raw bytes of the expanded public key. The hash is computed
-/// using Rpo256.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct PublicKey(Word);
+/// Public key represented as a polynomial with coefficients over the Falcon prime field.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PublicKey(pub Polynomial<FalconFelt>);
```

Comment:
Since we implement `Deref` into `Polynomial<FalconFelt>`, do we need to keep the inner field public?

Reply:
Makes sense, changed it now